synapse/lib/storm.py

```diff
@@ -918,3 +925,79 @@ def _stormOperDelTag(self, query, oper):
 
         for tag in tags:
             [ core.delTufoTag(node,tag) for node in nodes ]
+
+    def _stormOperJoinTags(self, query, oper):
+        args = oper[1].get('args')
+        opts = dict(oper[1].get('kwlist'))
+
+        nodes = query.data()
+
+        keep_nodes = opts.get('keep_nodes', False)
+        limt = LimitHelp(opts.get('limit'))
+
+        forms = {arg for arg in args}
+
+        tags = {tag for node in nodes for tag in s_tufo.tags(node, leaf=True)}
+
+        core = self.getStormCore()
+
```

Comment:
data model info is availible without actually asking the cortex storage layer:
```
if not forms:
    forms = [ name for (name,info) in core.getFormDefs() ]
```
and runtime has a helper for that:
```
for form,tag in self._iterPropTags(forms,tags)
```